spring-rabbit/src/test/java/org/springframework/amqp/rabbit/listener/JavaConfigFixedReplyQueueTests.java

```diff
@@ -129,6 +197,19 @@ public SimpleMessageListenerContainer serviceListenerContainer() {
 		}
 
 		/**
+		 * @return The reply listener container - the rabbit template with the wrong queue is the listener.
+		 */
+		@Bean
+		public SimpleMessageListenerContainer replyListenerContainerWrongQueue() {
+			SimpleMessageListenerContainer container = new SimpleMessageListenerContainer();
+			container.setConnectionFactory(rabbitConnectionFactory());
+			container.setQueues(replyQueue());
+			container.setMessageListener(fixedReplyQRabbitTemplateWrongQueue());
```

Comment:
I don't mind of your solution, but won't it looks better to override the `replyAddress` in the provided `ListenerContainerAware` with the value from this `queues`, instead of supporting the same value from both side?

Of course make a WARN if the `replyAddress` is presented already and isn't equal to the value from container.
From other side it looks bad to listen to the several queues, but use `RabbitTemplate` as a listener. The code there on the matter looks like:
```
PendingReply pendingReply = this.replyHolder.get(messageTag);
	if (pendingReply == null) {
		if (logger.isWarnEnabled()) {
			logger.warn("Reply received after timeout for " + messageTag);
		}
	throw new AmqpRejectAndDontRequeueException("Reply received after timeout");
}
```
But seems for me it isn't always true, because we may receive message from other queues on the listener container.

Have I missed anything?